deepforest/predict.py

```diff
@@ -140,10 +140,12 @@ def across_class_nms(predicted_boxes, iou_threshold=0.15):
     return new_df
 
 
-def predict_file(model,
-                 csv_file,
-                 root_dir,
+def predict_file(trainer,
+                 model,
+                 dataloader,
```

Comment:
I found this change a little confusing. `main.predict_file()` and `predict.predict_file()` now have different arguments and `predict.predict_file()` doesn't take a file as an argument. Assuming that the trainer and dataloader move was necessary as part of the broader refactor then I think we should consider renaming `predict.predict_file()` to make this easier to follow. `predict_single_file_dataloader` feels clunky, but something in this space to help communicate what's happening?

Reply:
A couple thoughts. The original function didn't do a good job isolating from main.DeepForest class methods, before you were literally passing the model object to use them to generate a dataloader.

https://github.com/weecology/DeepForest/blob/327ceb9094ec649f54822d88406b492c65b6851c/deepforest/predict.py#L173C1-L174C1

This doesn't make alot of sense. We want main.py to handle all within-class logic, and then hand off to predict.predict_* for tile, image or file prediction which is isolated from the class methods. I'm looking through docs right now, but I think the use should never use predict.predict_* methods directly. I think renaming is a good option 

predict.predict_tile = predict._predict_tile_ to highlight that these are internal function following 'private' names. https://peps.python.org/pep-0008/. I'm fine with renaming this predict._predict_dataloader_.

Reply:
Thanks for the explanation. Sounds good to me.

Reply:
just noticed that I don't know how to escape underscores in github markdown "predict._predict_image".